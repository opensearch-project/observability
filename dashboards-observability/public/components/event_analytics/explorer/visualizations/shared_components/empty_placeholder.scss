/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

 .lnsChart__empty {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  font-size: 20px;
}
