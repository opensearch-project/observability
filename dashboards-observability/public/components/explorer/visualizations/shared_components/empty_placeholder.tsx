/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiIcon, EuiText, IconType, EuiSpacer } from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';

export const EmptyPlaceholder = (props: { icon: IconType }) => (
  <>
    <EuiText className="lnsChart__empty" textAlign="center" color="subdued" size="xs">
      <EuiIcon type={props.icon} color="subdued" size="l" />
      <EuiSpacer size="s" />
      <p>
        <FormattedMessage
          id="xpack.lens.xyVisualization.noDataLabel"
          defaultMessage="No results found"
        />
      </p>
    </EuiText>
  </>
);
