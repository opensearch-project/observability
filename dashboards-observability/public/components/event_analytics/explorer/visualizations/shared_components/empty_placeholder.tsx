/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiIcon, EuiText, EuiSpacer } from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';
import './empty_placeholder.scss';

export const EmptyPlaceholder = (props: { icon: string }) => (
  <>
    <EuiText
      className="lnsChart__empty"
      textAlign="center"
      color="subdued"
      size="xs"
      data-test-subj="vizWorkspace__noData"
    >
      <EuiIcon type={props.icon} color="subdued" size="xxl" />
      <EuiSpacer size="l" />
      <p>
        <FormattedMessage id="visualization_noData" defaultMessage="No data found" />
      </p>
    </EuiText>
  </>
);
