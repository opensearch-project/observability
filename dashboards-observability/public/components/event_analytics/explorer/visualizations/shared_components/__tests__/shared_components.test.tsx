/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { EmptyPlaceholder } from '../empty_placeholder';
import { LensIconChartBar } from '../../../../../visualizations/assets/chart_bar';
import { ToolbarButton } from '../toolbar_button';

describe('Shared components', () => {
  configure({ adapter: new Adapter() });

  it('Renders empty placeholder component', async () => {
    const wrapper = mount(<EmptyPlaceholder icon={LensIconChartBar} />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders tool bar button component', async () => {
    const handleClick = jest.fn();
    const WrappedComponent = () => <div>testing</div>;

    const wrapper = mount(
      <ToolbarButton onClick={handleClick} fontWeight="bold">
        <WrappedComponent />
      </ToolbarButton>
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
