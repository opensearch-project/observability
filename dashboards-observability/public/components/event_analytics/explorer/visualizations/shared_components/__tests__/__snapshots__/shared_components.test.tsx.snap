// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared components Renders empty placeholder component 1`] = `
<EmptyPlaceholder
  icon={[Function]}
>
  <EuiText
    className="lnsChart__empty"
    color="subdued"
    data-test-subj="vizWorkspace__noData"
    size="xs"
    textAlign="center"
  >
    <div
      className="euiText euiText--extraSmall lnsChart__empty"
      data-test-subj="vizWorkspace__noData"
    >
      <EuiTextAlign
        textAlign="center"
      >
        <div
          className="euiTextAlign euiTextAlign--center"
        >
          <EuiTextColor
            color="subdued"
            component="div"
          >
            <div
              className="euiTextColor euiTextColor--subdued"
            >
              <EuiIcon
                color="subdued"
                size="xxl"
                type={[Function]}
              >
                <LensIconChartBar
                  aria-hidden={true}
                  className="euiIcon euiIcon--xxLarge euiIcon--subdued"
                  focusable="false"
                  role="img"
                  style={null}
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--xxLarge euiIcon--subdued"
                    fill="none"
                    focusable="false"
                    height={22}
                    role="img"
                    style={null}
                    viewBox="0 0 30 22"
                    width={30}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      className="lensChartIcon__subdued"
                      d="M5 7a1 1 0 011 1v13a1 1 0 01-1 1H1a1 1 0 01-1-1V8a1 1 0 011-1h4zm16-7a1 1 0 011 1v20a1 1 0 01-1 1h-4a1 1 0 01-1-1V1a1 1 0 011-1h4z"
                    />
                    <path
                      className="lensChartIcon__accent"
                      d="M13 11a1 1 0 011 1v9a1 1 0 01-1 1H9a1 1 0 01-1-1v-9a1 1 0 011-1h4zm16-7a1 1 0 011 1v16a1 1 0 01-1 1h-4a1 1 0 01-1-1V5a1 1 0 011-1h4z"
                    />
                  </svg>
                </LensIconChartBar>
              </EuiIcon>
              <EuiSpacer
                size="l"
              >
                <div
                  className="euiSpacer euiSpacer--l"
                />
              </EuiSpacer>
              <p>
                <FormattedMessage
                  defaultMessage="No data found"
                  id="visualization_noData"
                  values={Object {}}
                >
                  <span>
                    No data found
                  </span>
                </FormattedMessage>
              </p>
            </div>
          </EuiTextColor>
        </div>
      </EuiTextAlign>
    </div>
  </EuiText>
</EmptyPlaceholder>
`;

exports[`Shared components Renders tool bar button component 1`] = `
<ToolbarButton
  fontWeight="bold"
  onClick={[MockFunction]}
>
  <EuiButton
    className="lnsToolbarButton lnsToolbarButton--bold lnsToolbarButton--s"
    color="text"
    contentProps={
      Object {
        "className": "lnsToolbarButton__content",
      }
    }
    data-test-subj=""
    iconSide="right"
    iconType="arrowDown"
    onClick={[MockFunction]}
    size="s"
    textProps={
      Object {
        "className": "lnsToolbarButton__text",
      }
    }
  >
    <EuiButtonDisplay
      baseClassName="euiButton"
      className="lnsToolbarButton lnsToolbarButton--bold lnsToolbarButton--s"
      color="text"
      contentProps={
        Object {
          "className": "lnsToolbarButton__content",
        }
      }
      data-test-subj=""
      disabled={false}
      element="button"
      iconSide="right"
      iconType="arrowDown"
      isDisabled={false}
      onClick={[MockFunction]}
      size="s"
      textProps={
        Object {
          "className": "lnsToolbarButton__text",
        }
      }
      type="button"
    >
      <button
        className="euiButton euiButton--text euiButton--small lnsToolbarButton lnsToolbarButton--bold lnsToolbarButton--s"
        data-test-subj=""
        disabled={false}
        onClick={[MockFunction]}
        style={
          Object {
            "minWidth": undefined,
          }
        }
        type="button"
      >
        <EuiButtonContent
          className="euiButton__content lnsToolbarButton__content"
          iconSide="right"
          iconType="arrowDown"
          textProps={
            Object {
              "className": "euiButton__text lnsToolbarButton__text",
            }
          }
        >
          <span
            className="euiButtonContent euiButtonContent--iconRight euiButton__content lnsToolbarButton__content"
          >
            <EuiIcon
              className="euiButtonContent__icon"
              color="inherit"
              size="m"
              type="arrowDown"
            >
              <EuiIconEmpty
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                focusable="false"
                role="img"
                style={null}
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </EuiIconEmpty>
            </EuiIcon>
            <span
              className="euiButton__text lnsToolbarButton__text"
            >
              <WrappedComponent>
                <div>
                  testing
                </div>
              </WrappedComponent>
            </span>
          </span>
        </EuiButtonContent>
      </button>
    </EuiButtonDisplay>
  </EuiButton>
</ToolbarButton>
`;
